### PR TITLE
record TrialNames group by stages in ExperimentStatus

### PR DESCRIPTION
**Pull request by sperlingxx**
**What this PR does / why we need it**:
In this PR, several list fields are added to record TrialNames in different stages owned by current Experiment, which facilitates trials tracking. Count fields are removed since trial counts of each stage can be figured out by the length of TrialNames.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/1023)

### CODE REVIEW

**Comment by gaocegege** [comment]
Thanks for the PR. Not sure if we should do it, personally. Perfer to keep v1alpha3 API stable. I think we can add these new fields and keep the existing fields too.

WDYT @johnugeorge @hougangliu @richardsliu

**Comment by gaocegege** [comment]
/ok-to-test

@sperlingxx BTW, could you please add your infomation to https://github.com/kubeflow/katib/blob/master/docs/community/adopters.md if you are evaluating katib? Thanks

**Comment by sperlingxx** [comment]
> /ok-to-test
> 
> @sperlingxx BTW, could you please add your infomation to https://github.com/kubeflow/katib/blob/master/docs/community/adopters.md if you are evaluating katib? Thanks

It's done, please check this [PR](https://github.com/kubeflow/katib/pull/1025).

**Comment by sperlingxx** [comment]
> Thanks for the PR. Not sure if we should do it, personally. Perfer to keep v1alpha3 API stable. I think we can add these new fields and keep the existing fields too.
> 
> WDYT @johnugeorge @hougangliu @richardsliu

I also agree to keep v1alpha3 API stable, and I will restore count fields.

**Comment by johnugeorge** [comment]
Should we wait and move this to next api version(possible beta) bundled with other changes?

Related #906

**Comment by gaocegege** [comment]
@johnugeorge I think we can merge it since it does not breack v1alpha3 API.

If we have a breaking change in API, we should discuss it before we implement it in the new API.

WDYT @hougangliu @sperlingxx @johnugeorge

**Comment by hougangliu** [comment]
> @johnugeorge I think we can merge it since it does not break v1alpha3 API.
> 
> If we have a breaking change in API, we should discuss it before we implement it in the new API.
> 
> WDYT @hougangliu @sperlingxx @johnugeorge

SGTM

**Comment by hougangliu** [comment]
/approve
thanks

File: pkg/controller.v1alpha3/experiment/util/status_util.go [hunk 1]
```diff
@@ -50,26 +50,28 @@ func UpdateExperimentStatus(collector *ExperimentsCollector, instance *experimen
...
 
 	for index, trial := range trials.Items {
-		totalTrials++
+		sts.Trials++
 		if trial.IsKilled() {
-			trialsKilled++
+			sts.KilledTrialList = append(sts.KilledTrialList, trial.Name)
 		} else if trial.IsFailed() {
-			trialsFailed++
+			sts.KilledTrialList = append(sts.KilledTrialList, trial.Name)
```
Comment (andreyvelich) [inline pkg/controller.v1alpha3/experiment/util/status_util.go]: Should it be `sts.FailedTrialList` instead of `sts.KilledTrialList` ?

Comment (sperlingxx) [reply pkg/controller.v1alpha3/experiment/util/status_util.go]: @andreyvelich Yes, I've already fixed it.

**Comment by andreyvelich** [comment]
Thanks.
/lgtm

**Pull request merged**